### PR TITLE
User Input For Leaders

### LINKED ISSUE #6: Unnecessary Extra file in master branch- IntrotosubteamCalc.java
The changes that were necessary to be made  in the main class were made. This extra file is not necessary

### LINKED ISSUE #7: Compile Time Error in my Pull Request
A compile error occurs in #3 . Here is the error: 
```
/MyClass.java:54: error: variable leadersIncludedFinalCount might not have been initialized
          return leadersIncludedFinalCount;
                 ^
/MyClass.java:84: error: variable leadersIncludedOGCount might not have been initialized
          return leadersIncludedOGCount;
                 ^
2 errors
```
It says the two variables in the methods have not been initialized but they are initialized at the top of the method right here:
```java
      public static boolean userInputLeadersInOGCount() { //Returns whether leaders were included in original count; Made by Andrew
          boolean leadersIncludedOGCount;
```
And right here:
```java
      public static boolean userInputLeadersInFinalCount() { //Returns whether leaders will be included in final count; Made by Andrew
          boolean leadersIncludedFinalCount;
```
I might have made a stupid mistake and initialized it incorrectly.

### PR DESCRIPTION
**Pull request by Andrew-K961**
2 methods that ask the user about leaders.

Also closes #6 and #7

### CODE REVIEW

**Review by CooperW12 — DISMISSED** [review]

**Review by rahidas04 — DISMISSED** [review]

**Review by TheChar — CHANGES_REQUESTED** [review]
The return types for these methods should be boolean types, not Strings. You shouldn't need to print anything to the console except for the question. The outro method will take care of informing the user of that type of information.

**Comment by Andrew-K961** [comment]
> The return types for these methods should be boolean types, not Strings. You shouldn't need to print anything to the console except for the question. The outro method will take care of informing the user of that type of information.
@TheChar 

I can move the user feedback. The reason for using a string was due to the user responding with "yes" or "no" instead of "true" or "false". I can try to convert the string to a boolean with an if-statement.

**Review by GavenB141 — DISMISSED** [review]

**Review by TheChar — DISMISSED** [review]
This looks good! Please make sure the return type for userInputLeadersInOGCount() is set to boolean instead of String. Otherwise, it looks great!

File: MyClass.java [hunk 1]
```diff
@@ -22,5 +24,67 @@ public static int pot(){
...
+  
+    public static boolean userInputLeadersInFinalCount(){ //Returns whether leaders will be included in final count; Made by Andrew
+	boolean leadersIncludedFinalCount;
+        Scanner myObj = new Scanner(System.in);
+        
+        System.out.println("Are leaders included in the final count? (yes or no)");
+        String leaderIncludedFinalCount = myObj.nextLine();
+        
+        if (leaderIncludedFinalCount.equals("yes")){
+//            System.out.println("Leaders will be included in final count");
```
Comment (cyrusduong) [inline MyClass.java]: Not a requirement but a challenge:

These seem like a good message to print if you want a -v/--verbose mode to your program. Then the logging/output can be used to troubleshoot any problems.

Copy the following function snippet and paste at bottom of this class. Replace your commented out `println`s with this function use instead.
```
    public static void verboseLogging(String output){
        if (verboseMode == true){
            System.out.println(output);
        }
    }
```

Then, figure out how to read a `-v` [command-line argument](https://www.tutorialspoint.com/Java-command-line-arguments#:~:text=A%20command%2Dline%20argument%20is,array%20passed%20to%20main(%20)) when your program starts and use that to set a `verboseMode` boolean class variable when main is called.

> Hint: There's an `args[]` parameter variable in the `main(String args[])` function definition. See the link I posted above to use it.

File: MyClass.java [hunk 2]
```diff
@@ -22,5 +24,67 @@ public static int pot(){
...
+		leadersIncludedFinalCount = false;
+        }else{
+            System.out.println("Restart the program");
+        }
+        }
+        return leadersIncludedFinalCount;
+    }
+    
+    public static boolean userInputLeadersInOGCount(){ //Returns whether leaders were included in original count; Made by Andrew
+	    boolean leadersIncludedOGCount;
```
Comment (cyrusduong) [inline MyClass.java]: Fix indent. Should line up with Scanner variable.

File: MyClass.java [hunk 3]
```diff
@@ -22,5 +24,67 @@ public static int pot(){
...
+            if (leaderIncludedFinalCount.equals("yes")){
+//            System.out.println("Leaders will be included in final count");
+		    leadersIncludedFinalCount = true;
+	    }
+        else if (leaderIncludedFinalCount.equals("no")){
+//            System.out.println("Leaders will not be included in final count");
+		leadersIncludedFinalCount = false;
+        }else{
+            System.out.println("Restart the program");
+        }
```
Comment (cyrusduong) [inline MyClass.java]: I think the indenting here needs to be fixed?

File: MyClass.java [hunk 4]
```diff
@@ -22,5 +24,67 @@ public static int pot(){
...
+            if (leaderIncludedOGCount.equals("yes")){
+//            System.out.println("Leaders were included in original count");
+		    leadersIncludedOGCount = true;
+        }
+        else if (leaderIncludedOGCount.equals("no")){
+//            System.out.println("Leaders were not included in original count");
+		leadersIncludedOGCount = false;
+        }else{
+            System.out.println("Restart the program");
+        }
```
Comment (cyrusduong) [inline MyClass.java]: Same here. Think indents need fixed.

**Review by cyrusduong — DISMISSED** [review]
Small indent issues for readability. Fix the code indents and it will be good.
There's also a challenge if you wish to attempt it.

**Comment by cyrusduong** [comment]
For visibility @Andrew-K961 I've edited the description for GitHub to know to automatically close #6 when you merge this branch. This way people also don't grab the same issue and work on it on their branches.

**Comment by Andrew-K961** [comment]
> For visibility @Andrew-K961 I've edited the description for GitHub to know to automatically close #6 when you merge this branch. This way people also don't grab the same issue and work on it on their branches. @cyrusduong 

Thank you I didn't know how to do that.

**Review by TheChar — DISMISSED** [review]

**Review by GavenB141 — DISMISSED** [review]

**Review by Team1710CTO — DISMISSED** [review]

File: MyClass.java (hunk 1 above)
Comment (Andrew-K961) [reply MyClass.java]: I can implement this after this request gets merged.

**Review by TheChar — DISMISSED** [review]

**Review by GavenB141 — DISMISSED** [review]

**Review by TheChar — APPROVED** [review]

**Review by rahidas04 — APPROVED** [review]

**Review by GavenB141 — APPROVED** [review]

**Pull request merged**